### PR TITLE
Add list with OIDC authn supporting eventing components

### LINKED ISSUE #5857: Update eventing OIDC docs that EKB supports OIDC
Currently we list only that OIDC is supported in eventing-core with the MTChannelBasedBroker and InMemoryChannel.
Since we have OIDC support in eventing-kafka-broker (EKB) too, we can add a support matrix in the OIDC docs, showing for which Broker classes / Broker implementations and Channels OIDC is supported (currently for the MTChannelBasedBroker, eventing-kafka-broker, InMemoryChannel and KafkaChannel).

## In what release will this take happen (to the best of your knowledge)?
Ex. v1.14

## Context
Project dashboard: https://github.com/orgs/knative/projects/66

## Additional Information:
* Current OIDC documenation: https://knative.dev/docs/eventing/experimental-features/sender-identity/
* Eventing OIDC project: https://github.com/orgs/knative/projects/66/views/4
* EKB issues for OIDC: https://github.com/orgs/knative/projects/66/views/4?filterQuery=repo%3A%22knative-extensions%2Feventing-kafka-broker%22
* Useful links for Knative Docs style guide: https://github.com/knative/docs/blob/main/contribute-to-docs/README.md#style
* Overview of Knative Docs Contributor guide: https://github.com/knative/docs/blob/main/contribute-to-docs/README.md

### PR DESCRIPTION
**Pull request by creydr**
Fixes #5857

### CODE REVIEW

**Comment by creydr** [comment]
@pierDipi could you PTAL?

File: docs/eventing/experimental-features/sender-identity.md [hunk 1]
```diff
@@ -21,6 +21,21 @@ Knative Eventing Souces request an OIDC access token for the targets audience an
...
+## Compatibility
+
+OIDC authentication is currently supported for the following components:
+
+- Brokers:
+    - [MTChannelBasedBroker](./../brokers/broker-types/channel-based-broker/)
+    - [Knative Broker for Apache Kafka](./../brokers/broker-types/kafka-broker/)
+- Channels:
+    - [InMemoryChannel](https://github.com/knative/eventing/tree/release-1.13/config/channels/in-memory-channel/README.md)
+    - [KafkaChannel](https://github.com/knative-extensions/eventing-kafka-broker/tree/release-1.13/README.md)
```
[the remark this answers is not in the document]
Comment (creydr) [reply docs/eventing/experimental-features/sender-identity.md]: Ah ok. I just took the links from https://knative.dev/docs/eventing/channels/channels-crds/. Updated it...

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**